services/web/server/src/simcore_service_webserver/projects/projects_utils.py

```diff
@@ -286,3 +294,69 @@ def any_node_inputs_changed(
                     )
                     return True
     return False
+
+
+def get_frontend_node_outputs_changes(
+    new_node: dict[str, Any], old_node: dict[str, Any], filter_keys: set[ServiceKey]
+) -> FrontendOutputsChanges:
+    """if node is a specific type it checks if outputs changed"""
+    nodes_keys = {old_node.get("key"), new_node.get("key")}
+    if not (len(nodes_keys) == 1 and nodes_keys.pop() in filter_keys):
+        return FrontendOutputsChanges(changed=False, keys=set())
+
+    log.debug("Comparing nodes %s %s", new_node, old_node)
+    outputs_changed = new_node.get("outputs") != old_node.get("outputs")
+
+    def _get_outputs_keys(node_data: dict[str, Any]) -> set[str]:
+        outputs = node_data.get("outputs", {})
+        if outputs is None:
+            return set()
+        return set(outputs.keys())
+
+    changed_keys = _get_outputs_keys(new_node).symmetric_difference(
+        _get_outputs_keys(old_node)
+    )
+    return FrontendOutputsChanges(changed=outputs_changed, keys=changed_keys)
+
+
+def find_changed_dict_keys(
+    current_dict: dict[str, Any],
+    new_dict: dict[str, Any],
+    *,
+    look_for_removed_keys: bool,
+) -> dict[str, Any]:
+    # The `store` key inside outputs can be either `0` (integer) or `"0"` (string)
+    # this generates false positives.
+    # Casting to `str` to fix the issue.
+    # NOTE: this could make services relying on side effects to stop form propagating
+    # changes to downstream connected services.
+    # Will only fix the issue for `file-picker` to avoid issues.
+    def _cast_outputs_store(dict_data: dict[str, Any]) -> None:
+        for data in dict_data.get("outputs", {}).values():
+            if "store" in data:
+                data["store"] = int(data["store"])
+
+    if current_dict.get("key") == "simcore/services/frontend/file-picker":
```

Comment:
sorry, this implementation feel very hacky provided the generic name (``find_changed_dict_keys``) you gave to the function! Make sure it is only used in your specific case.

Reply:
So I think the comment in the function is explaining what the issue is. I'd guess that we need a better name for the function since it was moved from its original module. Changing to `find_changed_node_keys`